pkg/controller/elasticsearch/client/model.go

```diff
@@ -264,6 +286,14 @@ func (l License) IsValid(instant time.Time) bool {
 		l.ExpiryTime().After(instant)
 }
 
+// IsSupported return true if the current license type is supported by the given version of Elasticsearch.
+func (l License) IsSupported(v *version.Version) bool {
+	if l.Type == string(ElasticsearchLicenseTypeEnterprise) && !v.IsSameOrAfter(version.MustParse("7.6.0")) {
+		return false
+	}
+	return true
```

Comment:
I know this statement can be collapsed into the just returning directly from the comparison, but I found the intent much easier to understand by being a bit redundant here.